include/rfb.js

```diff
@@ -572,9 +577,21 @@ var RFB;
             }
         },
 
-        _handleKeyPress: function (keysym, down) {
+        _handleKeyPress: function (keyevent) {
             if (this._view_only) { return; } // View only, skip keyboard, events
-            RFB.messages.keyEvent(this._sock, keysym, down);
+
+            var down = (keyevent.type == 'keydown');
+
+            if (this._QEMU_Extended_Key_Event) {
+                var xt_scancode = this._XT_scancode[keyevent.code];
+                if (xt_scancode) {
+                    var keysym = keyevent.keysym;
+                    RFB.messages.QEMUExtendedKeyEvent(this._sock, keysym, down, xt_scancode);
+                }
```

Comment:
I believe we can log a message here saying that we couldn't find the xt_scancode for the given keyevent.code